qa/demo.py

```diff
@@ -0,0 +1,277 @@
+        print("\nSC wallet balance in satoshi: {}".format(
+            json.dumps(sc_balance, indent=4, sort_keys=True)))
+
+        boxes_balances = sc_node.wallet_allBoxes()["result"]
+        print("\nSC wallet boxes: {}".format(json.dumps(boxes_balances, indent=4, sort_keys=True)))
+
+
+        # Do BT
+        self.pause()
+        mc_address = self.nodes[0].getnewaddress("")
+        bt_amount = 2  # Zen
+        withdrawal_request = {"outputs": [
+            {"publicKey": mc_address,
+             "value": bt_amount * coin}
+        ]
+        }
+
+        print("\nCreating Backward Transfer request to withdraw {} Zen to the Mainchain...".format(bt_amount))
+        sc_node.transaction_withdrawCoins(json.dumps(withdrawal_request))
+
+        print "Generating SC Block with Backward Transfer request transaction..."
+        scblock_id3 = generate_next_blocks(sc_node, "first node", 1)[0]
+
+        self.pause()
+
+        # Run block generation till epoch end -> automatic block generation
+        print("Generating 9 more MC blocks to finish withdrawal epoch for the Sidechain...")
+        mc_block_ids = mc_node.generate(9)
+        print "MC Block ids - {}\n".format(mc_block_ids)
+        print("Generating SC blocks to synchronize MC blocks and automatically start creation of Withdrawal Certificate...")
+        sc_block_ids = generate_next_blocks(sc_node, "first node", 4)
+        print("\nGenerating Withdrawal Certificate...\n")
+
+        attempts = 20
+        while mc_node.getmempoolinfo()["size"] == 0 and attempts > 0:
+            if attempts % 4 == 0:
+                print("Wait for withdrawal certificate in MC memory pool...")
+            time.sleep(10)
+            attempts -= 1
+            sc_node.block_best()  # just a ping to SC node. For some reason, STF can't request SC node API after a while idle.
+        assert_equal(1, mc_node.getmempoolinfo()["size"], "Certificate was not added to Mc node mmepool.")
+
+        certHash = mc_node.getrawmempool()[0]
+        print("Withdrawal certificate hash - " + certHash)
+        cert = mc_node.getrawcertificate(certHash, 1)
+        print("Withdrawal certificate - {}".format(json.dumps(cert, indent=4, sort_keys=True, default=str)))
+        # TODO: check mc balances before and after block inclusion
+
+        self.pause()
+
+
+        # Gen SC block to sync back certificate
+        print("\nGenerating 1 more MC block to include Withdrawal certificate in the chain...")
+        mc_block_4 = mc_node.generate(1)[0]
+        print "MC Block id - {}\n".format(mc_block_4)
+
+        self.pause()
+
+        # Get SC balances changes
+        sc_balance = sc_node.wallet_balance()["result"]
+        print("\nSC wallet balance in satoshi: {}".format(
+            json.dumps(sc_balance, indent=4, sort_keys=True)))
+        boxes_balances = sc_node.wallet_allBoxes()["result"]
+        print("\nSC wallet boxes: {}".format(json.dumps(boxes_balances, indent=4, sort_keys=True)))
```

Comment:
verify mc address balance is considering BT

Reply:
done